R/capture-requests.R

```diff
@@ -260,24 +260,46 @@ dbGetInfoConTrace <- quote({
   }
 })
 
-dbGetInfoResultTrace <- quote({
-  thing <- returnValue()
-  # TODO: would this be better if we traced the methods individually?
-  if (inherits(dbObj, c("MariaDBResult", "PqResult", "SQLiteResult"))) {
-    hash <- hash(dbObj@sql)
-  } else if (inherits(dbObj, "OdbcResult")) {
-    hash <- hash(dbObj@statement)
+#' Extract a hash from a (result) object
+#'
+#' This function should generally not be used, but must be exported for the
+#' query recording function to work properly
+#'
+#' @param obj the DBI result object to hash
+#'
+#' @return hash for the object
+#' @export
```

Comment:
This exporting was the critical bit, because of the way that the tracing works, we need to export these functions for them to be available in the search space at that point in time.